languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,107 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="QUELLE_QUE" name="quelle que" default="temp_off">
+          <rule>
+              <pattern>
+                  <token postag="N f s" skip="-1"/>
+                  <marker>
+                      <token regexp="yes">quel|quels|quelles|qu(&apostrophe;)</token>
+                      <token min="0">elle</token>
+                  </marker>
+                  <token regexp="yes">qu(&apostrophe;)</token>
+                  <token>elle</token>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelle</suggestion> ?</message>
+              <example correction="quelle">On trouvera la solution, <marker>qu'elle</marker> qu'elle soit.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token regexp="yes">qu(&apostrophe;)</token>
+                  <token>elle</token>
+                  <token regexp="yes">qu(&apostrophe;)</token>
+                  <token>un</token>
+              </antipattern>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">quel|quels|quelles|qu(&apostrophe;)</token>
+                      <token min="0">elle</token>
+                  </marker>
+                  <token regexp="yes" skip="-1">que|qu(&apostrophe;)</token>
+                  <token min="0">en</token>
+                  <token>soit</token>
+                  <token postag="D f s|D e s" postag_regexp="yes"/>
+                  <token postag="N f s"/>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelle</suggestion> ?</message>
+              <example correction="quelle">Il le fera <marker>qu'elle</marker> que soit sa situation.</example>
+              <example correction="quelle">Il le fera <marker>qu'elle</marker> qu'en soit la raison.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">quelle|quel|quels|qu(&apostrophe;)</token>
+                      <token regexp="yes" min="0">elle|elles</token>
+                  </marker>
+                  <token regexp="yes" skip="1">que|qu(&apostrophe;)</token>
+                  <token regexp="yes">en|elles</token>
+                  <token>soient</token>
+                  <token postag="D f p|D e p" postag_regexp="yes"/>
```

Comment:
Please revisit every instance of this regex. The system will interpret this as `D f p e p` or `D f D e p`. My recommendation to go for more "elegant" ways of expressing this (e.g. `D [ef] p`) is not just for cosmetic reasons, it also prevents this kind of pitfalls.